### PR TITLE
Generify Listeners

### PR DESCRIPTION
**Pull request by ZeroMemes**
This will actually make it possible to use comms as a library and not as a drag+drop source and modify under MIT codebase.

### CODE REVIEW

File: src/main/java/cabaletta/comms/Pipe.java [hunk 1]
```diff
@@ -74,11 +74,11 @@ public iMessage receiveMessage() throws IOException {
 
         @Override
         public void close() {
-            closed = true; // Prevent further messages from being sent and received
+            Pipe.this.closed = true; // Prevent further messages from being sent and received
```
Comment (leijurv) [inline src/main/java/cabaletta/comms/Pipe.java:77]: this is cring

Comment (ZeroMemes) [reply src/main/java/cabaletta/comms/Pipe.java:77]: It's easier to read, makes it more explicit that the closed flag resides in the parent class.

**Review by leijurv — APPROVED** [review]
sure

File: src/main/java/cabaletta/comms/Pipe.java (hunk 1 above)
Comment (0-x-2-2) [reply src/main/java/cabaletta/comms/Pipe.java:77]: this is based

**Review by 0-x-2-2 — APPROVED** [review]
😻

**Pull request merged**